internal/util/prometheus.go

```diff
@@ -0,0 +1,55 @@
+package util
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+type ControllerFunctionalPrometheusMetrics struct {
+	// number of post /config to proxy successfully
+	ConfigPassCounter prometheus.Counter
+	// number of post /config to proxy failed
+	ConfigFailureCounter prometheus.Counter
```

Comment:
Optimally we'd have only one metric for the event of performing proxy configuration, where success would be distinguished from failure by labels.

Consider creating only one metric, called e.g. `proxy_configuration_count` with labels:
- `success` with possible values `true`, `false`
- `type` with possible values `deck`, `post-config`

Reply:
Sounds Good.